### PR TITLE
Restrict simultaneous usage of host/hostname/port options fixes #105

### LINKED ISSUE #105: The client initialization did not verify the hostname
```
const { connect } = require('socketcluster-client')
const socket = connect({
  // hostname: 'localhost',
  port: 3333
})
```
In node env, the instance `socket` above with init no hostname, so the hostname will be undefined, but the instance will be create sucess. I don't think this is a good way, I suggest
- Valid the `hostname` option item
-  Set default value as 'localhost'
- Throw exception

### PR DESCRIPTION
**Pull request by MegaGM**
Otherwise `hostname` is `undefined` if you run the client via Node.js

### CODE REVIEW

**Comment by MegaGM** [comment]
The only one usecase of `port` option without `hostname` option, which I see atm for Node.js client, is when you're in dev environment. So I guess we could restrict simultaneous usage of host/hostname/port options with no issues.

1. We either will use `hostname` + `port`, or `host` alone. Not together. Then we could compute `host` once only, inside `.create()`. It will allow us to get rid from computing host inline in some functions like there https://github.com/SocketCluster/socketcluster-client/blob/master/lib/scsocketcreator.js#L28-L31 hence simplify `socketcluster-client` codebase abit.

2. From the other hand, thinking about simplicity for users, the client has a default value for `hostname` option, the same hostname as in current URI. What if we simply set `localhost` as a default value for Node.js?
Otherwise socketcluster-client's API will be inconsistent :P You'll be able to subscribe without `hostname` when in a browser, but won't be able do the same in Node.js

I'll propose changes for 1 in meantime.

**Comment by jondubois** [comment]
@MegaGM well described, I might even prefer your solution 2 - To use localhost as the default. What do you think?

Sometimes people use the client to communicate with SC from a different application running on the same host so that would be one use case.

**Comment by MegaGM** [comment]
Let's start with 2. Then I'll push some changes for 1, and if they aren't good but evil, I'll revert the commit.

**Comment by MegaGM** [comment]
Is it okay to check the hostname:port *that way*?
If you want to merge, I'll change the PR's title. If you don't, I'll revert the last commit and change title anyway :smile_cat:

File: lib/scsocketcreator.js [hunk 1]
```diff
@@ -58,6 +58,7 @@ function create(options) {
   var opts = {
     port: getPort(options, isSecureDefault),
     hostname: global.location && location.hostname || '',
```
Comment (MegaGM) [inline lib/scsocketcreator.js]: I noticed I've added a new string, will fix it.

File: lib/scsocketcreator.js [hunk 2]
```diff
@@ -47,17 +47,23 @@ function create(options) {
...
+  if (options.host && !options.host.match(/[^:]+:\d{2,5}/)) { // should be "hostname:port"
+    throw new InvalidArgumentsError('The host option should include both' +
+      ' the hostname and the port number in the format "hostname:port".');
+  }
+
+  if ((options.host && options.port) || (options.host && options.hostname)) {
+    throw new InvalidArgumentsError('The host option should already include' +
+      ' the hostname and the port number in the format "hostname:port"' +
+      ' - Because of this, the host and port options cannot be specified together.' +
+      ' Use either hostname and port options together or host option alone.');
```
Comment (jondubois) [inline lib/scsocketcreator.js]: @MegaGM Maybe this should be two separate if statements with two different error messages?
The part `the host and port options cannot be specified together` is not relevant if the user passed `host` and `hostname`.

Comment (jondubois) [reply lib/scsocketcreator.js]: Also, you'll probably think this is silly nit picking (and you'd be right ;p) but the convention in the rest of the code currently is that error strings are a single sentence without any dot characters - See https://github.com/SocketCluster/socketcluster/blob/master/index.js#L181-L183 - Although I actually like it more with a dot at the end (it's more correct grammatically), I think that it's more important to be consistent.

Comment (MegaGM) [reply lib/scsocketcreator.js]: > the host and port options cannot be specified together

Overlooked. My bad.

Comment (MegaGM) [reply lib/scsocketcreator.js]: DRY, sir. So 1 `if` is better than 2, I guess, since they are tightly related.

**Pull request merged**